lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -205,6 +205,39 @@ static Node *simplifyNode(Node *node, Function *F) {
   return node;
 }
 
+// 2D vector of dimentions mapping a dimention that is reshaped into either 1
+// or 2 dimentions by the first reshape node in a ChannelShuffle operation
+using ChannelShuffleDimGroups =
+    llvm::SmallVector<llvm::SmallVector<unsigned_t, 2>, max_tensor_dimensions>;
+
+/// \returns the dims of \p node grouped by which input dimention they came from
```

Comment:
Oops I'll fix the description here to say returns success/failure and populates dimGroups with the output